snuba/datasets/transactions.py

```diff
@@ -137,10 +135,14 @@ def transactions_migrations(
     ]
     for col_name in low_cardinality_cols:
         col = current_schema.get(col_name)
-        if col and not col.column_type.startswith("LowCardinality"):
-            new_type = f"LowCardinality({col.column_type})"
+
+        if col and LowCardinality not in col.get_all_modifiers():
+            if isinstance(col, WithDefault):
+                col.inner_type = LowCardinality(col.inner_type)
+            else:
+                col = LowCardinality(col)
```

Comment:
Why not just apply the alter without doing any check ?
It should be a noop if the column is already in the right format. But I would verify on Clickhouse if it complains bout this.

Reply:
Do you mean making this potentially `LowCardinality(LowCardinality(String()))`? That won't work.

Reply:
No, nevermind, I thought about running something like 
```
ALTER TABLE table MODIFY COLUMN {col_name} {col_type}
```
where col_type is the formatted expression from the Snuba schema (the source of truth), not from the current schema. But I realized you do not have access to the snuba schema here.